doc/west/repo-tool.rst

```diff
@@ -0,0 +1,266 @@
+manifest repository. Running ``west init`` without specifying a manifest
+repository will use `a default manifest repository
+<https://github.com/zephyrproject-rtos/manifest>`_. A different manifest
+repository can be selected with ``west init --manifest-url URL``.
+
+The format of the West manifest is described by a `pykwalify
+<https://pypi.org/project/pykwalify/>`_ schema, found `here
+<https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/meta/west/commands/schema.yml>`_.
+
+The ``manifest-rev`` branch
+***************************
+
+Each project managed through the multi-repo commands has a ``manifest-rev``
+branch that points to the project's revision as specified in the manifest. The
+``manifest-rev`` branch is updated whenever project data is fetched (see the
+`command overview`_ below).
+
+All branches created through the multi-repo commands track the ``manifest-rev``
+branch. Several multi-repo commands also use ``manifest-rev`` as a reference
+for the upstream revision (as of the most recent fetch).
+
+.. note::
+
+   ``manifest-rev`` is a normal Git branch, and is only treated specially by
+   name. If you delete it or otherwise modify it, it will be
+   recreated/reset when upstream data is next fetched by ``west``, as if
+   through ``git reset --hard`` (though ``git update-ref`` is used internally).
+
+   Since ``manifest-rev`` represents the upstream revision as of the most
+   recent fetch, it is normally a bad idea to modify it.
+
+   ``manifest-rev`` was added to allow branches to track SHA revisions, and to
+   give a consistent reference for the upstream revision regardless of how the
+   manifest changes over time.
+
+Command overview
+================
+
+This section gives a quick overview of the multi-repo commands, split up by
+functionality. All commands loosely mimic the corresponding Git command, but in
+a multi-repo context (``west fetch`` fetches upstream data without updating
+local work branches, etc.)
+
+Passing no projects to commands that accept a list of projects usually means to
+run the command for all projects listed in the manifest.
+
+.. note::
+
+   For the most up-to-date documentation, see the command help texts (e.g.
+   ``west fetch --help``). Only the most important flags are mentioned here.
+
+Cloning and updating projects
+*****************************
+
+After running ``west init`` to initialize a West installation, the following
+commands will clone/update projects.
+
+Except for ``west rebase``, all commands in this section implicitly update the
+manifest repository, fetch upstream data, and update the ``manifest-rev``
+branch.
+
+- ``west clone [-b BRANCH_NAME] [PROJECT ...]``: Clones the specified
+  projects (default: all projects).
+
+  An initial branch named after the project's revision is created in each
+  cloned repository. A different name can be specified with ``-b``.
+
+  Like all branches created by the multi-repo commands, the initial branch
```

Comment:
"Through West" maybe, to make it clear that it's talking about branches created by the user.

The talk about multi-repo commands is because we're bundling a bunch of other things in West as well, but it might be clear enough anyway.